src/mono/mono/component/CMakeLists.txt

```diff
@@ -79,17 +82,53 @@ set(${MONO_DIAGNOSTICS_TRACING_COMPONENT_NAME}-dependencies
   ${MONO_DIAGNOSTICS_TRACING_COMPONENT_NAME}-gen-sources
 )
 
+# marshal-ilgen
+list(APPEND components
+  ${MONO_MARSHAL_ILGEN_COMPONENT_NAME}
+)
+list(APPEND components_for_aot
+  ${MONO_MARSHAL_ILGEN_COMPONENT_NAME}
+)
+
+set(${MONO_MARSHAL_ILGEN_COMPONENT_NAME}-sources
+  ${MONO_COMPONENT_PATH}/marshal-ilgen.c
+  ${MONO_COMPONENT_PATH}/marshal-ilgen.h
+  ${MONO_COMPONENT_PATH}/marshal-ilgen-noilgen.c
+  ${MONO_COMPONENT_PATH}/marshal-ilgen-noilgen.h  
```

Comment:
I'm having a real hard time wrapping my head around what happens here with the `NOT ENABLE_ILGEN` setting. 

We build a single component that works whether or not `libmono-ilgen.a` is linked in?

But I don't get how the component is linkable if we the final executable ends up not including the `libmono-ilgen.a` library.  I get that everything in `marshal-ilgen.c` now does an indirect call back into the runtime.  But isn't the runtime always initializing that callbacks list?

Reply:
We solved this by making ENABLE_ILGEN always true.